docs/_dapp/abigen.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Abigen
+sort-key: p
+---
+
+Abigen is a binding-generator for easily interacting with Ethereum using Go. 
+Abigen creates easy-to-use, type-safe Go packages from Ethereum smart contract 
+definitions known as ABIs. This abstracts away a lot of the complexity of handling 
+smart contract deployment and interaction in Go native applications such as 
+encoding and decoding events. Abigen comes bundled with 
+Geth. Abigen can also be built 
+independently by navigating to `go-ethereum/cmd/abigen` and running `go build`, or 
+equivalently:
+
+```sh
+$ cd $GOPATH/src/github.com/ethereum/go-ethereum
+$ go build ./cmd/abigen
+```
+
+## What is an ABI?
+
+Ethereum smart contracts have a schema that defines its functions and return types 
+in the form of a JSON file. This JSON file is known as an _Application Binary Interface_, 
+or ABI. The ABI acts as a specification for precisely how to encode data sent to a 
+contract and how to decode the data the contract sends back. The ABI is the only 
+essential piece of information required to generate Go bindings. Go developers can then 
```

Comment:
Actually, binary code generated by compiler is already required in cases that binding with Deployment function is expected.

Reply:
```suggestion
essential piece of information required to generate Go bindings, unless a deployment function is expected in which case the contrat bytecode is also needed. Go developers can then 
```